src/python/pants/backend/graph_info/tasks/filter.py

```diff
@@ -45,7 +45,7 @@ def __init__(self, *args, **kwargs):
     def _get_targets(spec_str):
       spec_parser = CmdLineSpecParser(get_buildroot())
       try:
-        address_spec = spec_parser.parse_address_spec(spec_str)
+        address_spec = spec_parser.parse_spec(spec_str)
```

Comment:
This will probably blow up if the wrong type comes out of this union? The error message would not be horrific, but could be better.

You might consider leaving the `parse_address_spec` method, and having it blow up nicely if it can't be parsed as that type? These legacy v1 codepaths will continue to not support filesystem specs, I expect.

Reply:
I looked into how this task is implemented, and it looks like the `spec_str` is being populated from `context.target_roots`, which is guaranteed to be address specs. So, this should not be an issue. 

Hopefully, this task will be ported very soon to V2, also, which is why I didn't originally spend much time making this file more robust.

Generally, I looked at adding back `parse_address_spec` and I couldn't find a way to deduplicate it and `parse_spec`. (Because the target shortcut is the last block in the conditionals, we need the FS logic to happen before that, so it doesn't mix well.)